pkg/reconciler/managed/reconciler.go

```diff
@@ -607,7 +606,7 @@ func (r *Reconciler) Reconcile(_ context.Context, req reconcile.Request) (reconc
 			// condition. If not, we requeue explicitly, which will trigger
 			// backoff.
 			log.Debug("Cannot remove managed resource finalizer", "error", err)
-			managed.SetConditions(xpv1.ReconcileError(err))
+			managed.SetConditions(xpv1.Deleting(), xpv1.ReconcileError(err))
 			return reconcile.Result{Requeue: true}, errors.Wrap(r.client.Status().Update(ctx, managed), errUpdateManagedStatus)
 		}
 
```

Comment:
I think we need to set `xpv1.Creating()` in line 638 as well, which is called `if meta.ExternalCreateIncomplete(managed)`

Reply:
Line 699 as well, in `if !observation.ResourceExists && meta.ExternalCreateSucceededDuring(managed, r.creationGracePeriod)`

Reply:
Line 793, where `if err := r.client.Update(ctx, managed); err != nil`